src/services/APIServiceBase.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * This is the base class for all services that interact with the API system.
+ */
+
+import { Validation } from "io-ts";
+import * as msRest from "ms-rest-js";
+import * as winston from "winston";
+import { ReadableReporter } from "../utils/validation_reporters";
+
+const messageErrorOnUnknownResponse = "Unknown response.";
+const messageErrorOnApiError = "Api error.";
+
+export default class APIServiceBase {
```

Comment:
I don't see why these methods should stay in a class since they don't share any state with the extended class, they can be just pure methods, can't they?